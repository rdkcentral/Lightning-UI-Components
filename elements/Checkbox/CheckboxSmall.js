import Checkbox from './Checkbox';
import * as styles from './CheckboxSmall.styles';
import { withExtensions } from '../../mixins';

class CheckboxSmall extends Checkbox {
  static get __themeStyles() {
    return styles;
  }
}

export default withExtensions(CheckboxSmall);
