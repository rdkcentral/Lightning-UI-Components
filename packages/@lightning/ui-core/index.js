export { default as utils } from './src/utils/index.js';
export * from './src/components/index.js';
export * from './src/globals/index.js';
export * from './src/mixins/index.js';
export * from './src/shaders/index.js';
export * from './src/textures/index.js';
