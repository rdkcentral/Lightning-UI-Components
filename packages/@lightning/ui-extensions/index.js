export * from './src/FocusRing';
export * from './src/DropShadow';
