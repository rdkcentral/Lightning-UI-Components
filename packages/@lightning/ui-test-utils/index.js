export * from './src/lightning-test-renderer';
export * from './src/lightning-test-utils';
export * from './src/lng-test-env';
