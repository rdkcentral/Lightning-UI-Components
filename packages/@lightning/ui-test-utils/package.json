{
  "name": "@lightning/ui-test-utils",
  "version": "1.1.0",
  "type": "module",
  "exports": "./index.js",
  "files": [
    "src/*"
  ],
  "peerDependencies": {
    "@lightningjs/core": "^2.7.1"
  },
  "devDependencies": {
    "jest": "^29.3.1",
    "jest-environment-jsdom": "^29.3.1",
    "jest-webgl-canvas-mock": "^0.2.3"
  },
  "dependencies": {
    "@lightning/ui-core": "workspace:^",
    "@lightningjs/core": "^2.7.1"
  }
}
