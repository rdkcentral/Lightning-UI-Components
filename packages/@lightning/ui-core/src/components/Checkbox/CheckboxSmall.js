import Checkbox from './Checkbox.js';
import * as styles from './CheckboxSmall.styles.js';
import { withExtensions } from '../../mixins/index.js';

class CheckboxSmall extends Checkbox {
  static get __themeStyle() {
    return styles;
  }
}

export default withExtensions(CheckboxSmall);
