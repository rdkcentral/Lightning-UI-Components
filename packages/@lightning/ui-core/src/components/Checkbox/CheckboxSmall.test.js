import { makeCreateComponent } from '@lightning/ui-test-utils';
import CheckboxSmall from './CheckboxSmall.js';

const createCheckboxSmall = makeCreateComponent(CheckboxSmall);

describe('CheckboxSmall', () => {
  let checkbox, testRenderer;

  beforeEach(() => {
    [checkbox, testRenderer] = createCheckboxSmall({ checked: false });
  });

  it('renders', () => {
    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct height', () => {
    expect(checkbox.h).toBe(20);
  });

  it('renders the correct weight', () => {
    expect(checkbox.w).toBe(20);
  });
});
