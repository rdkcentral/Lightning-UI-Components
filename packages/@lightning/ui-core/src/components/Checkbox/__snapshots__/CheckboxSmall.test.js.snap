// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxSmall renders 1`] = `
{
  "Component": {
    "Check": {
      "active": false,
      "alpha": 0,
      "attached": true,
      "boundsMargin": null,
      "clipping": false,
      "color": 4294967295,
      "enabled": false,
      "flex": false,
      "flexItem": false,
      "h": 10,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0.5,
      "mountX": 0.5,
      "mountY": 0.5,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "Check",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": [
        "Check",
      ],
      "type": "Icon",
      "visible": true,
      "w": 8,
      "x": [Function],
      "y": [Function],
      "zIndex": 0,
    },
  },
}
`;
